src/ORM/FieldType/DBComposite.php

```diff
@@ -206,16 +191,14 @@ public function setValue($value, $record = null, $markChanged = true)
     }
 
     /**
-     * Bind this field to the dataobject, and set the underlying table to that of the owner
-     *
-     * @param DataObject $dataObject
+     * Bind this field to the model, and set the underlying table to that of the owner
      */
-    public function bindTo($dataObject)
+    public function bindTo(DataObject $dataObject): void
```

Comment:
Unlike the places where I've gone for `ViewableData`, this one does seem to need to be `DataObject` explicitly. There's a conditional check elsewhere that only calls this method if the record is an instance of `DataObject`.